src/debuggability.js

```diff
@@ -128,6 +130,68 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+function registerPromise() {
+    if (Promise.monitor) {
+        Promise.monitor._promiseIdCounter++;
+        this._promiseId = Promise.monitor._promiseIdCounter;
+        Promise.monitor._pendingPromises[Promise.monitor._promiseIdCounter]
+            = this;
+    }
+}
+
+function unregisterPromise() {
+    if (Promise.monitor)
+        delete Promise.monitor._pendingPromises[this._promiseId];
+}
+
+function enableMonitoring () {
+    if (!Promise.monitor) {
+        // Property that holds monitoring related info,
+        // existence of it means that monitoring feature is currently enabled
+        Promise.monitor = {};
+        Promise.monitor._pendingPromises = {};
```

Comment:
in this file somewhere, like the other closure variables. Or is it used from other files?